drivers/espi/espi_mchp_xec.c

```diff
@@ -520,6 +527,103 @@ static int espi_xec_receive_oob(struct device *dev,
 	return 0;
 }
 
+static int espi_xec_flash_read(struct device *dev,
+			       struct espi_flash_packet *pckt)
+{
+	int ret;
+	struct espi_xec_data *data = (struct espi_xec_data *)(dev->driver_data);
+	u32_t err_mask = MCHP_ESPI_FC_STS_IBERR |
+			MCHP_ESPI_FC_STS_FAIL |
+			MCHP_ESPI_FC_STS_OVFL |
+			MCHP_ESPI_FC_STS_BADREQ;
+
+	LOG_DBG("%s", __func__);
+
+	if (!(ESPI_FC_REGS->STS & MCHP_ESPI_FC_STS_CHAN_EN)) {
+		LOG_ERR("Flash channel is disabled");
+		return -EIO;
+	}
+
+	if (pckt->len > MAX_SLAVE_BUFFER_SIZE) {
+		LOG_ERR("Invalid size request");
+		return -EINVAL;
+	}
+
+	ESPI_FC_REGS->FL_ADDR_MSW = 0;
+	ESPI_FC_REGS->FL_ADDR_LSW = pckt->flash_addr;
+	ESPI_FC_REGS->MEM_ADDR_MSW = 0;
+	ESPI_FC_REGS->MEM_ADDR_LSW = (u32_t)&slave_mem[0];
+	ESPI_FC_REGS->XFR_LEN = pckt->len;
+	ESPI_FC_REGS->CTRL = MCHP_ESPI_FC_CTRL_FUNC(MCHP_ESPI_FC_CTRL_RD0);
+	ESPI_FC_REGS->CTRL |= MCHP_ESPI_FC_CTRL_START;
+
+	/* Wait until ISR or timeout */
+	ret = k_sem_take(&data->flash_lock, K_MSEC(MAX_FLASH_TIMEOUT));
+	if (ret == -EAGAIN) {
+		LOG_ERR("%s timeout", __func__);
+		return -ETIMEDOUT;
+	}
+
+	if (ESPI_FC_REGS->STS & err_mask) {
```

Comment:
@scottwcpg In case of an error,  do we have to clear the Status ?  I was checking with Alberto and this register is R/WC

Reply:
yes, you must clear the r/w-1-clear bits.

Reply:
Corrected in latest version